supervision/detection/tools/smoother.py

```diff
@@ -115,4 +115,8 @@ def get_smoothed_detections(self) -> Detections:
             if track is not None:
                 tracked_detections.append(track)
 
-        return Detections.merge(tracked_detections)
+        detections = Detections.merge(tracked_detections)
+        if len(tracked_detections) == 0:
```

Comment:
```suggestion
        if len(detections) == 0:
```

Reply:
Hi @LinasKo 👋🏻 ! Great find! That's probably the only change I'd make.

Reply:
Done :+1:

Also verified the `__len__` implementation just in case. Seems to be alright - can't imagine changes to `Detections.empty` that would break this.

Reply:
I'm considering modifying the `Detections.empty` API to include arguments such as `with_confidence`, `with_tracker_id`, etc. So that in the future, in a similar situation, we would only need to do `Detections.empty(with_tracker_id = True)`.

Reply:
The problem with that (#943, Option 4) is that intermediary functions that would also need `with_confidence` and `with_tracker_id`, or you need to forbid them from ever accepting zero-length iterables.

We're actually at the best place to show this.

If nothing is detected, `detections = Detections.merge(tracked_detections)` takes `[]`. It then tries to call `Detections.empty` - but what params should it set?..